### ISSUE TITLE
Multiple screen support

### ISSUE THREAD

**Issue by sindresorhus** [sindresorhus/Plash#2]
I intentionally left it out in the initial version as I don't use multiple screens myself, so I'm not exactly sure how it should optimally work. **Please leave some feedback if you need multiple screen support.**

Possible solutions:

1. Add preference to display it on the second screen if connected.
2. Add preference to display it on a selected screen if connect. Would be a dropdown to choose screen.

I'm open to other ideas.

**Comment by firrae** [comment]
I think you covered the primary options needed. Personally I'd just put the same thing on both screens, but I could also see someone wanting different pages per screen. That might be useful, though likely a bit more to implement.

**Comment by ianiv** [comment]
It would be useful for me to have a different URL per screen

**Comment by lacking-utility** [comment]
Another vote for different url per screen or drop down to set screen. I’ve got four monitors connected and would primarily like to put a calendar on one desktop, and maybe weather on another

**Comment by jakearchibald** [comment]
I currently have a different desktop background per screen per desktop, so I'd want this to behave the same.

**Comment by sindresorhus** [comment]
Plash v1.1.0 adds the ability to at least choose what display to show the website wallpaper on.

**Comment by jcousins-ynap** [comment]
I would appreciate multiple monitor support. I can see how it would make the preferences panel more complicated, not to mention how the URL is chosen.

**Comment by jhavens12** [comment]
I would also appreciate multiple monitor support - I use 3 displays and currently can only get the background on one display. A different webpage per display would be ideal. Thanks

**Comment by sindresorhus** [comment]
This will happen eventually, but there are other issues I plan to prioritize first. Contributions are always welcome though, both in form of figuring out how it should work and also code.

**Comment by jcousins-ynap** [comment]
@sindresorhus Is this the app that uses the infamous NSMenu?

**Comment by sindresorhus** [comment]
@jcousins-ynap Are you referring this [tweet](https://twitter.com/sindresorhus/status/1186957040494604288)?

**Comment by jcousins-ynap** [comment]
@sindresorhus I am ;)

**Comment by sindresorhus** [comment]
@jcousins-ynap That tweet is about [Dato](https://sindresorhus.com/dato). Plash does use NSMenu, but only as a static menu, so I won't be encountering any of those problems here.

**Comment by long-lazuli** [comment]
> Plash v1.1.0 adds the ability to at least choose what display to show the website wallpaper on.

I think an option to set "all monitors" would be a good next step :)

**Comment by sindresorhus** [comment]
I'm unfortunately unlikely to work on this anytime soon. I only use a single screen myself, so it's simply not a priority.

However, I thought of a workaround today. You can run multiple instances of Plash at the same time. One for each display. And hide the extra menu bar icons with something like [Bartender](https://www.macbartender.com/) (paid), [hidden](https://github.com/dwarvesf/hidden) (free), or [Dozer](https://github.com/Mortennn/Dozer) (free).

Here are additional builds of Plash with a different bundle ID so they can individual preferences:

1. Plash in the App Store.
2. [Plash2.app.zip](https://github.com/sindresorhus/Plash/files/4874984/Plash2.app.zip)
3. [Plash3.app.zip](https://github.com/sindresorhus/Plash/files/4874987/Plash3.app.zip)
4. [Plash4.app.zip](https://github.com/sindresorhus/Plash/files/4874989/Plash4.app.zip)

**Comment by sooxt98** [comment]
@sindresorhus Im not able to run Plash on external monitor.

**Comment by sindresorhus** [comment]
@sooxt98 This is not the place to report a bug. Open a new issue with more info.

**Comment by tigger04** [comment]
also +1 vote for this, but understand you've got to prioritize.
thanks for the workaround, came here to find this :)

**Comment by wesleycoder** [comment]
> I'm unfortunately unlikely to work on this anytime soon. I only use a single screen myself, so it's simply not a priority.
> 
> However, I thought of a workaround today. You can run multiple instances of Plash at the same time. One for each display. And hide the extra menu bar icons with something like [Bartender](https://www.macbartender.com/) (paid), [hidden](https://github.com/dwarvesf/hidden) (free), or [Dozer](https://github.com/Mortennn/Dozer) (free).
> 
> Here are additional builds of Plash with a different bundle ID so they can have individual preferences:
> 
> 1. Plash in the App Store.
> 2. [Plash2.app.zip](https://github.com/sindresorhus/Plash/files/4874984/Plash2.app.zip)
> 3. [Plash3.app.zip](https://github.com/sindresorhus/Plash/files/4874987/Plash3.app.zip)
> 4. [Plash4.app.zip](https://github.com/sindresorhus/Plash/files/4874989/Plash4.app.zip)

Could the updates for these different bundle IDs be automated on your build pipeline?
These versions are outdated now and I wonder how can we keep them updated.

**Comment by sindresorhus** [comment]
@wesleycoder I'm unlikely to spend time on automating a hacky workaround. If I'm gonna spend time on something, it would be a proper solution. However, I have updated the builds in https://github.com/sindresorhus/Plash/issues/2#issuecomment-653891524 to the latest version of Plash.

**Comment by alicehamberger** [comment]
for mac, I definately use several "Desktops" (Backgrounds) and so having several pluck bagrounds would be perfect!
That way I can put my google calendar and a google sheets on two sepparate backgrounds!

**Comment by sindresorhus** [comment]
I have updated the builds to the latest Plash version.

**Comment by communque** [comment]
+1 - Plash is a genuinely superior alternative to the built-in mac desktop.  That it currently allows JS explodes the possibilities in ways not only "cool" but genuinely useful and productive.  Building multiple screen support would be my #1 request.  Thanks for the great work.

**Comment by Diavolezza** [comment]
+1
support of multiple screens would be great.
Anyway, Plash is top!

**Comment by laurencebush** [comment]
Please please please make it so we can have 8 different Plashes running at once, I use all 4 currently and would like to use more.

**Comment by sindresorhus** [comment]
@laurencebush I have updated the builds and added 8 now.

**Comment by StashPowerUser** [comment]
Would it be possible to get updated builds for 2.15?

**Comment by ELLIOTTCABLE** [comment]
Two related requests, one tiny and related to these builds, and one that could easily be included in the primary build:

1. when hovering in the menubar, these builds all say "Plash", without the ident digit — could that be fixed w/ plist changes or sth?

2. Ideally, the drop-down could show the loaded URL or some other description, whether or not one is running multiple copies in parallel …

**Comment by legndery** [comment]
> Please please please make it so we can have 8 different Plashes running at once, I use all 4 currently and would like to use more.

Is it for separate displays? cause for me when I open. multiple instances of Plash it just overlays them in single screen.

**Comment by sindresorhus** [comment]
> Is it for separate displays? cause for me when I open. multiple instances of Plash it just overlays them in single screen.


You can choose the screen in the app settings.

**Comment by legndery** [comment]
> > Is it for separate displays? cause for me when I open. multiple instances of Plash it just overlays them in single screen.
> 
> You can choose the screen in the app settings.

I am opening using standard `open -n -a "Plash"`, but the settings feels shared, like whenever I change monitor in one, it changes in the other instance. More proof is, I add one website in one instance, it auto adds in other instance. I tried with using different versions but no luck.

**Comment by sindresorhus** [comment]
You are opening multiple instances of the same app, not the different apps provided. As documented, they have different names and bundle IDs.

**Comment by astout** [comment]
I found a nice workaround for multi-monitor support. I've created a `plash-cloner` script that enables the end user to create their own clone of Plash.app. This is similar to the distributed bundles offered by @sindresorhus, but relieves the author of the burden of maintaining multiple bundle IDs. End user is responsible for keeping their clones up to date when the official Plash.app is updated.

- https://github.com/astout/plash-cloner

----

_Thank you @sindresorhus for this useful piece of software 🙏._

**Comment by JHenzi** [comment]
> I found a nice workaround for multi-monitor support. I've created a `plash-cloner` script that enables the end user to create their own clone of Plash.app. This is similar to the distributed bundles offered by [@sindresorhus](https://github.com/sindresorhus), but relieves the author of the burden of maintaining multiple bundle IDs. End user is responsible for keeping their clones up to date when the official Plash.app is updated.
> 
> * https://github.com/astout/plash-cloner
> 
> _Thank you [@sindresorhus](https://github.com/sindresorhus) for this useful piece of software 🙏._

This works. The instructions had me in the CLI, but then told me to double click. My Mac wouldn't let me open the command, but could execute it from the shell.

Flightaware under one desktop, MagicMirror under the other.

**Comment by lxpk** [comment]
Bump +1 I would also like multiple monitor support without using cloning solutions. It would be most excellent for this to be built in and I would donate to support it.

**Comment by lxpk** [comment]
+1 again, I know project is no longer open source but I would make this in a closed fork if it helps it get done faster, love Plash.

**Comment by wwerkk** [comment]
Bump. Would love this feature!

**Comment by 45Matches** [comment]
> I think you covered the primary options needed. Personally I'd just put the same thing on both screens, but I could also see someone wanting different pages per screen. That might be useful, though likely a bit more to implement.

Plus one for different websites on each screen.
